cunumeric/runtime.py

```diff
@@ -267,7 +267,7 @@ def get_next_random_epoch(self):
         return result
 
     def is_supported_type(self, dtype):
-        return np.dtype(dtype) in self.legate_context.type_system
+        return dtype in self.legate_context.type_system
```

Comment:
to accept this change, I want you to make sure that all callsites of `is_supported_type` indeed pass an `np.dtype` object and nothing else. this is something mypy can help, but until we put type annotations, we'd need to do this manually.

Reply:
then you need to make sure that `dtype` can only be either a string of form "PointN" or an np.dtype object.

for future reference, I highly recommend you make proper dtypes for all PointN types instead of using strings, though you don't have to do that in this PR. while I was chasing the non-deterministic bug with advanced indexing, it was annoying that I couldn't print out the output of the zip task. (I worked around it by rewriting the type to np.int64, but that's only doable for 1d cases.) giving them proper dtypes with the right binary layout would be useful for debugging later.

Reply:
Sounds good. I have added the check and created an issue https://github.com/nv-legate/cunumeric/issues/385